blaze/mahimahi/server/server.py

```diff
@@ -85,6 +128,26 @@ def start_server(
                     log.warn("skipping", file_name=file.file_name, method=file.method, uri=file.uri, host=file.host)
                     continue
 
+                # if this is the root html file, then we want to insert our snippet here
+                # to extract critical requests.
+                # TODO: do we need to do for all other HTML pages, not just root?
+                if file.headers.get("content-type", None) == "text/html" and file.uri == '/':
```

Comment:
It's not necessary that the "root" page is `/` (i.e. the user loaded a page that is not the root of a page) -- this ties into your previous question: we should jsut do it for all HTML pages. If the element is not present in the HAR, then it'll just be ignored anyway.